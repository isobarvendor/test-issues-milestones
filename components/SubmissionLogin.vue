<template>
  <div class="login-box column center mechanics">
    <p class="text-center" v-html="submissionText.siginHeader"></p>
    <a
      class="social-button fb"
      id="login_fb"
      v-on:click="redirectTo('facebook')"
      v-if="social.facebook"
      ><i class="ic ic-fb" /><span
        >{{ submissionText.continue }} Facebook</span
      ></a
    >
    <a
      class="social-button line"
      v-on:click="redirectTo('line')"
      v-if="social.line"
      ><i class="ic ic-line" /><span
        >{{ submissionText.continue }} LINE</span
      ></a
    >
    <a
      class="social-button google"
      id="login_google"
      v-on:click="redirectTo('google')"
      v-if="social.google"
      ><i class="ic ic-google" /><span
        >{{ submissionText.continue }} Google</span
      ></a
    >
    <a class="social-button email" href="/login" v-if="social.email"
      ><i class="ic ic-email" /><span
        >{{ submissionText.continue }} email</span
      ></a
    >
    <div v-if="errorMessage" class="error">
      {{ errorMessage }}
    </div>
  </div>
</template>

<script>
import { translation } from "@/constants/index";
export default {
  name: "Login",
  props: {
    data: null,
    social: null
  },
  data() {
    return {
      submissionText: translation.submissionText
    };
  },
  methods: {
    redirectTo(source) {
      location.href =
        "/api/oauth2/authorize/" +
        source +
        "?redirect_uri=" +
        window.location.origin +
        "/settoken";
    }
  },
  mounted() {
//     if (location.protocol !== 'https:') {
//     location.replace(`https:${location.href.substring(location.protocol.length)}`);
// }
  },
  computed: {
    errorMessage() {
      return this.$store.state.errorLogin;
    }
  },
  mounted() {}
};
</script>

<style scoped>
.text-center {
  text-align: center;
}
.error {
  color: #000;
  margin-top: 30px;
}
</style>
