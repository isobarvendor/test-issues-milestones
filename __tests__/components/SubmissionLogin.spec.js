
import { shallowMount, createLocalVue } from '@vue/test-utils'

import SubmissionLogin from "@/components/SubmissionLogin";

import Vuex from 'vuex';


let localVue, store;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    store = new Vuex.Store({
      state:{
        errorLogin: "tes"
      }
    });
  });

const factory = () => {
  return shallowMount(SubmissionLogin, {
    localVue, store,
    propsData: {
      social:{
        line:true,
        facebook:true,
        google:true,
      }
    }
  });
};

describe("SubmissionLogin", () => {

  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
