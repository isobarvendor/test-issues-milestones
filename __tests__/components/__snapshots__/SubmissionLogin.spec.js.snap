// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionLogin renders properly 1`] = `
<div class="login-box column center">
  <p class="text-center">Sign in or register a Coca-Cola <br>account to enter the draw</p> <a class="social-button fb"><i class="ic ic-fb"></i><span>Continue with Facebook</span></a> <a class="social-button line"><i class="ic ic-line"></i><span>Continue with LINE</span></a> <a class="social-button google"><i class="ic ic-google"></i><span>Continue with Google</span></a>
  <!---->
  <div class="error">
    tes
  </div>
</div>
`;
